src/components/leaderboards/Leaderboard/index.js

```diff
@@ -77,62 +82,73 @@ module.exports = React.createClass({
     });
   }, 100, { trailing: true }),
 
-  renderLeaderboardItems: function() {
-    if (this.state.isLoading) {
-      return <Icon className="Leaderboard__loading" icon="refresh" />;
-    }
+  renderLoadingState: function() {
+    return <Icon className="Leaderboard__loading" icon="refresh" />;
+  },
+
+  renderEmptyState: function() {
+    var emptyText       = this.t('emptyText');
+    var emptyButtonText = this.t('emptyButtonText');
+
+    return <LeaderboardEmpty emptyText={ emptyText } emptyButtonText={ emptyButtonText } { ...this.props } />;
+  },
 
+  renderLeaderboardItems: function() {
     var currentPage = this.state.currentPage - 1;
     var board = this.state.boardData[currentPage];
 
-    if (!board) return;
-
-    return board.map(function(d,i) {
-      var formattedAmount = this.formatAmount(d.amount);
-      var formattedRank = numeral(d.rank).format('0o');
-
-      return (
-        <LeaderboardItem
-          key={ d.id }
-          rank={ formattedRank }
-          name={ d.name }
-          url={ d.url }
-          isoCode={ d.isoCode }
-          amount={ formattedAmount }
-          imgSrc={ d.medImgSrc }
-          width={ this.state.childWidth }
-          renderImage={ this.props.renderImage } />
-      );
-    }, this);
+    return (
+      <ReactCSSTransitionGroup
+        className="Leaderboard__items"
+        transitionName="Leaderboard__animation"
+        component="ol">
+          {
+            board.map(function(d, i) {
```

Comment:
Good point – that function was written a long time ago (I've just moved it around in this PR) but I could make it clearer